go.mod

```diff
@@ -19,6 +19,7 @@ require (
 	github.com/giantswarm/micrologger v0.5.0
 	github.com/google/go-cmp v0.5.4
 	github.com/stretchr/testify v1.7.0
+	gomodules.xyz/jsonpatch/v2 v2.0.1
```

Comment:
I haven't been able to automatically generate patches with our current `github.com/evanphx/json-patch` library.

Also, this new one is the same library [used by `controller-runtime` to generate patches](https://github.com/kubernetes-sigs/controller-runtime/blob/e1a725df2743147795e5dfc8275365f7ada24805/pkg/webhook/admission/response.go#L22).